app/controllers/incomeSources/add/BusinessAddedObligationsController.scala

```diff
@@ -0,0 +1,167 @@
+                                                   incomeSourceDetailsService: IncomeSourceDetailsService,
+                                                   nextUpdatesService: NextUpdatesService,
+                                                   val obligationsView: BusinessAddedObligations)
+                                                  (implicit val appConfig: FrontendAppConfig,
+                                                   implicit val itvcErrorHandlerAgent: AgentItvcErrorHandler,
+                                                   implicit override val mcc: MessagesControllerComponents,
+                                                   val ec: ExecutionContext,
+                                                   dateService: DateServiceInterface)
+  extends ClientConfirmedController with I18nSupport with FeatureSwitching{
+
+  lazy val backUrl: String = controllers.incomeSources.add.routes.AddBusinessReportingMethod.show().url
+  private lazy val agentBackUrl = controllers.incomeSources.add.routes.AddBusinessReportingMethod.showAgent().url
+
+  def show(id: Option[String]): Action[AnyContent] = (checkSessionTimeout andThen authenticate andThen retrieveNino
+    andThen retrieveIncomeSources andThen retrieveBtaNavBar).async {
+    implicit user =>
+      handleRequest(isAgent = false, id)
+  }
+
+  def showAgent(id: Option[String]): Action[AnyContent] =
+    Authenticated.async {
+      implicit request =>
+        implicit user =>
+          getMtdItUserWithIncomeSources(incomeSourceDetailsService) flatMap {
+            implicit mtdItUser =>
+              handleRequest(isAgent = true, id)
+          }
+    }
+
+  def handleRequest(isAgent: Boolean, id: Option[String])(implicit user: MtdItUser[_], ec: ExecutionContext): Future[Result] = {
+    if (isDisabled(IncomeSources)) {
+      Future.successful(Redirect(controllers.routes.HomeController.show()))
+    } else {
+      id match {
+        case None => Logger("application").error(
+          s"[BusinessAddedObligationsController][handleRequest] - Error: No id supplied by reporting method page")
+          Future(itvcErrorHandler.showInternalServerError())
+        case Some(value) =>
+          val addedBusiness: BusinessDetailsModel = user.incomeSources.businesses.find(x => x.incomeSourceId.contains(value)).get
+
+          if (addedBusiness.tradingName.isEmpty) {
+            Logger("application").error(
+              s"[BusinessAddedObligationsController][handleRequest] - No business name for business with provided id")
+            itvcErrorHandler.showInternalServerError()
+          }
+          if (addedBusiness.tradingStartDate.isEmpty) {
+            Logger("application").error(s"[BusinessAddedObligationsController][handleRequest] - No business start date for business with provided id")
+            Future(itvcErrorHandler.showInternalServerError())
+          }
+
+          val businessName: String = addedBusiness.tradingName.get
```

Comment:
again, calling .get is not safe way in we encourage to avoid this was of coding

Reply:
The if statement above handles the case where the name isn't defined and throws an error, so the get is safe. I understand if it's still bad coding practice and you'd like me to change it